src/components/layout/Search/SearchArea.js

```diff
@@ -1,20 +1,26 @@
 import React, { useContext } from "react";
 import RecipeContext from "../../../context/recipes/RecipeContext";
 import IngredientForm from "./IngredientForm";
+import IngredientList from "./IngredientList";
 
 import styles from "./SearchArea.module.css";
-import IngredientList from "./IngredientList";
 
-const { ingredientSearchContainer, searchBtn } = styles;
+const { ingredientSearchContainer, searchBtn, searchBtnDisabled } = styles;
 
 const SearchArea = () => {
-  const { getRecipes } = useContext(RecipeContext);
+  const { getRecipes, ingredientList, isLoading } = useContext(RecipeContext);
 
   return (
     <div className={ingredientSearchContainer}>
       <IngredientForm />
       <IngredientList />
-      <button onClick={getRecipes} className={searchBtn}>
+      <button
+        onClick={getRecipes}
+        className={`${searchBtn} ${
+          ingredientList.length === 0 || isLoading ? searchBtnDisabled : ""
+        }`}
+        disabled={ingredientList.length === 0 || isLoading}
```

Comment:
On `line 20` and `line 22`, you can run into an error if the `ingredientList` variable is ever `null`. We would get an error that may lead to unexpected results when we do `ingredientList.length` if `ingredientList` is null. That is because we cannot do `.length` on a `null` object.

To protect against this potential issue, we use something called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). Optional chaining is a JavaScript operator that checks if the variable value is not `null` before accessing a property from it (the length property in our case).

So, the way it applies here is by doing something like `ingredientList?.length` on both `line 20` and `line 22`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining